src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -211,6 +228,21 @@ export class SharedCredentialsProvider implements CredentialsProvider {
             })
     }
 
+    private makeSsoProvider() {
+        // These properties are validated before reaching this method
+        const ssoRegion = this.profile[SHARED_CREDENTIAL_PROPERTIES.SSO_REGION]!
+        const ssoUrl = this.profile[SHARED_CREDENTIAL_PROPERTIES.SSO_START_URL]!
+
+        const ssoOidcClient = new AWS.SSOOIDC({ region: ssoRegion })
```

Comment:
Should we explicitly set anonymous credentials for this client?

Reply:
For reference:

- jetbrains / java way here: https://github.com/aws/aws-toolkit-jetbrains/blob/731c03d318e5d06b621a53a6817f73e6[…]s/core/credentials/profiles/ProfileCredentialProviderFactory.kt
- Did not find an explicit way to do this in https://github.com/aws/aws-sdk-js/issues , https://github.com/aws/aws-sdk-js-v3